tests/test_usm_type.py

```diff
@@ -1416,6 +1416,14 @@ def test_histogram_bin_edges(usm_type_v, usm_type_w):
     assert edges.usm_type == du.get_coerced_usm_type([usm_type_v, usm_type_w])
 
 
+@pytest.mark.parametrize("usm_type", list_of_usm_types, ids=list_of_usm_types)
+def test_select(usm_type):
+    condlist = [dp.array([True, False], usm_type=usm_type)]
+    choicelist = [dp.array([1, 2], usm_type=usm_type)]
+    res = dp.select(condlist, choicelist)
+    assert res.usm_type == usm_type
```

Comment:
We have to test different combination of `usm_type` per each input:
```suggestion
@pytest.mark.parametrize("usm_type_x", list_of_usm_types, ids=list_of_usm_types)
@pytest.mark.parametrize("usm_type_y", list_of_usm_types, ids=list_of_usm_types)
def test_select(usm_type_x, usm_type_y):
    condlist = [dp.array([True, False], usm_type=usm_type_x)]
    choicelist = [dp.array([1, 2], usm_type=usm_type_y)]
    res = dp.select(condlist, choicelist)
    assert res.usm_type == du.get_coerced_usm_type([usm_type_x, usm_type_y])
```